sway-core/src/error.rs

```diff
@@ -187,6 +191,26 @@ where
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Hash)]
+pub struct Hint {
+    msg: Option<String>,
+}
+
+impl Hint {
+    pub fn empty() -> Hint {
+        Hint { msg: None }
+    }
+    pub fn new(msg: String) -> Hint {
+        Hint { msg: Some(msg) }
+    }
+}
```

Comment:
This doesn't seem very useful, but I assume you've made it just to implement `Display` so the `#[error ...]` attribute works below?  Feels overengineered but I can't think of a better solution, other than to get `Option` support working automatically in `#[error]` somehow.

Reply:
>I assume you've made it just to implement Display so the #[error ...] attribute works below?

Yes.